tests/test_controller_host.py

```diff
@@ -0,0 +1,141 @@
+
+        self._network = network
+
+    def tearDown(self):
+        self._network.stop(True)
+
+    def test_instantiation(self):
+        self.assertEqual(self.controller_host.host_id, "host_1")
+        self.assertEqual(self.controller_host.computing_host_ids, ["QPU_1"])
+        self.assertEqual(self.controller_host.get_operation_execution_time("QPU_1", "SINGLE", "X"), 1)
+
+        self.controller_host.add_computing_host_to_network("QPU_2")
+        self.assertEqual(self.controller_host.computing_host_ids, ["QPU_1", "QPU_2"])
+        self.assertEqual(self.controller_host.get_operation_execution_time("QPU_1", "REC_ENT", None), 1)
+
+    def test_distributed_scheduler(self):
+        self.controller_host.add_computing_host_to_network("QPU_2")
+
+        q_map = {
+            'qubit_1': 'QPU_1',
+            'qubut_2': 'QPU_1',
+            'qubit_3': 'QPU_2',
+            'qubit_4': 'QPU_2'}
+
+        # Form layer 1
+        op_1 = Operation(
+            name="SINGLE",
+            qids=["qubit_1"],
+            gate="H",
+            computing_host_ids=["QPU_1"])
+
+        op_2 = Operation(
+            name="SEND_ENT",
+            qids=["qubit_2", "qubit_3"],
+            computing_host_ids=["QPU_1", "QPU_2"])
+
+        op_3 = Operation(
+            name="REC_ENT",
+            qids=["qubit_3", "qubit_2"],
+            computing_host_ids=["QPU_2", "QPU_1"])
+
+        layer_1 = Layer([op_1, op_2, op_3])
+
+        # Form layer 2
+        op_1 = Operation(
+            name="TWO_QUBIT",
+            qids=["qubit_3", "qubit_4"],
+            gate="cnot",
+            computing_host_ids=["QPU_2"])
+
+        layer_2 = Layer([op_1])
+
+        # Form layer 3
+        op_1 = Operation(
+            name="MEASURE",
+            qids=["qubit_3"],
+            cids=["bit_1"],
+            computing_host_ids=["QPU_2"])
+
+        layer_3 = Layer([op_1])
+
+        # Form layer 4
+        op_1 = Operation(
+            name="SEND_CLASSICAL",
+            cids=["bit_1"],
+            computing_host_ids=["QPU_2", "QPU_1"])
+
+        op_2 = Operation(
+            name="REC_CLASSICAL",
+            cids=["bit_1"],
+            computing_host_ids=["QPU_1", "QPU_2"])
+
+        layer_4 = Layer([op_1, op_2])
+
+        # Form layer 5
+        op_1 = Operation(
+            name="CLASSICAL_CTRL_GATE",
+            qids=["qubit_1"],
+            cids=["bit_1"],
+            gate="X",
+            computing_host_ids=["QPU_1"])
+
+        layer_5 = Layer([op_1])
+
+        layers = [layer_1, layer_2, layer_3, layer_4, layer_5]
+        circuit = Circuit(q_map, layers)
+
+        computing_host_schedules = self.controller_host.create_distributed_schedule(circuit)
+
+        self.assertEqual(len(computing_host_schedules), 2)
+        self.assertEqual(len(computing_host_schedules['QPU_1']), 4)
+        self.assertEqual(len(computing_host_schedules['QPU_2']), 4)
+
+        self.assertEqual(computing_host_schedules['QPU_1'][0]['name'], "SINGLE")
```

Comment:
Might want to make a separate file with a class that contains all of the constants like Gates.SINGLE = "SINGLE"

Reply:
Ah.. seems like you've got this already...